### PR TITLE
build(examples): separate examples from library project

### LINKED ISSUE #16: Split examples from real library projects
## Description

At the moment the examples are mixed in different ways directly into the project itself:
1) as sub-project for the ktor-example
2) directly inside the tests of the core-library (see e.g. https://github.com/stuebingerb/KGraphQL/blob/cc6efaeb821d3075480caaf374ce21797309f799/kgraphql/src/test/kotlin/com/apurebase/kgraphql/server/NettyServer.kt)

This has drawbacks:
* dependency updates in the examples do not need to be released or be mentioned in the release-notes (who cares?)
* when publishing the dependency-graph _all_ dependencies only used in the examples can safely be excluded
  * this easy for the ktor-example project, one can just exclude this sub-project
  * it's harder for the example directly in the test-sources of the core-library
* it's not easy to see which dependencies are only there for the examples

### Separate root-project inside the same repository

Create a new root-project named "examples", that contains all examples.
The core-library can be included using [Composite Builds](https://docs.gradle.org/current/userguide/composite_builds.html), to depend on the current code-state inside the repository instead of a published version.

### Move examples to separate ``KGraphQL-examples` repository

Create a whole new project for the examples, that contains all examples.
This way the examples can only depend on actually released versions of the library though, but this isn't necessarily a bad thing anyways.

### PR DESCRIPTION
**Pull request by mervyn-mccreight**
Fix #16

### CODE REVIEW

File: examples/ktor/src/main/resources/application.conf [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ktor {
   deployment {
     port = 8080
-    port = ${?PORT}
```
Comment (stuebingerb) [inline examples/ktor/src/main/resources/application.conf:4]: Looks like standard configuration to support environment variables, what was the reason for removal?

**Comment by stuebingerb** [comment]
Maybe an addition to the README as to how to build the examples (since they are no longer build with the main project)?

**Comment by mervyn-mccreight** [comment]
Good idea, will add something

File: examples/ktor/src/main/resources/application.conf (hunk 1 above)
Comment (mervyn-mccreight) [reply examples/ktor/src/main/resources/application.conf:4]: It did not seem to be necessary to me for an example on how to use the ktor integration, but I can put it back in if you prefer that.

Comment (stuebingerb) [reply examples/ktor/src/main/resources/application.conf:4]: Fine for me, I was just wondering 😉

**Review by stuebingerb — APPROVED** [review]

**Pull request merged**